source/gameengine/Ketsji/KX_CubeMap.cpp

```diff
@@ -0,0 +1,248 @@
+*/
+
+/** \file KX_CubeMap.cpp
+*  \ingroup ketsji
+*/
+
+#include "KX_CubeMap.h"
+#include "KX_GameObject.h"
+#include "KX_Globals.h"
+#include "KX_BlenderSceneConverter.h"
+
+#include "RAS_Texture.h"
+
+#include "DNA_texture_types.h"
+
+KX_CubeMap::KX_CubeMap(KX_BlenderSceneConverter *converter, KX_GameObject *gameobj, RAS_Texture *texture, RAS_IRasterizer *rasty)
+	:RAS_CubeMap(texture, rasty),
+	m_viewpointObject(gameobj),
+	m_invalidProjection(true),
+	m_ignoreLayers(0),
+	m_clipStart(0.0f),
+	m_clipEnd(0.0f),
+	m_autoUpdate(true),
+	m_forceUpdate(true)
+{
+	MTex *mtex = m_texture->GetMTex();
+
+	EnvMap *env = mtex->tex->env;
+	if (env->object) {
+		KX_GameObject *obj = converter->FindGameObject(env->object);
+		if (obj) {
+			m_viewpointObject = obj;
+		}
+	}
+
+	m_ignoreLayers = env->notlay;
+
+	m_clipStart = env->clipsta;
+	m_clipEnd = env->clipend;
+
+	m_autoUpdate = (env->flag & ENVMAP_AUTO_UPDATE) != 0;
+
+	m_texture->SetCubeMap(this);
+}
+
+KX_CubeMap::~KX_CubeMap()
+{
+	m_texture->SetCubeMap(NULL);
+}
+
+STR_String& KX_CubeMap::GetName()
+{
+	return m_texture->GetName();
+}
+
+KX_GameObject *KX_CubeMap::GetViewpointObject() const
+{
+	return m_viewpointObject;
+}
+
+void KX_CubeMap::SetViewpointObject(KX_GameObject *gameobj)
+{
+	m_viewpointObject = gameobj;
+}
+
+void KX_CubeMap::SetInvalidProjectionMatrix(bool invalid)
+{
+	m_invalidProjection = invalid;
+}
+
+bool KX_CubeMap::GetInvalidProjectionMatrix() const
+{
+	return m_invalidProjection;
+}
+
+void KX_CubeMap::SetProjectionMatrix(const MT_Matrix4x4& projection)
+{
+	m_projection = projection;
+}
+
+const MT_Matrix4x4& KX_CubeMap::GetProjectionMatrix() const
+{
+	return m_projection;
+}
+
+int KX_CubeMap::GetIgnoreLayers() const
+{
+	return m_ignoreLayers;
+}
+
+float KX_CubeMap::GetClipStart() const
+{
+	return m_clipStart;
+}
+
+float KX_CubeMap::GetClipEnd() const
+{
+	return m_clipEnd;
+}
+
+void KX_CubeMap::SetClipStart(float start)
+{
+	m_clipStart = start;
+}
+
+void KX_CubeMap::SetClipEnd(float end)
+{
+	m_clipEnd = end;
+}
+
+bool KX_CubeMap::NeedUpdate()
+{
+	bool result = m_autoUpdate || m_forceUpdate;
+	m_forceUpdate = false;
+
+	return result;
+}
+
+#ifdef WITH_PYTHON
+
+PyTypeObject KX_CubeMap::Type = {
+	PyVarObject_HEAD_INIT(NULL, 0)
+	"KX_CubeMap",
+	sizeof(PyObjectPlus_Proxy),
+	0,
+	py_base_dealloc,
+	0,
+	0,
+	0,
+	0,
+	py_base_repr,
+	0, 0, 0, 0, 0, 0, 0, 0, 0,
+	Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE,
+	0, 0, 0, 0, 0, 0, 0,
+	Methods,
+	0,
+	0,
+	&CValue::Type,
+	0, 0, 0, 0, 0, 0,
+	py_base_new
+};
+
+PyMethodDef KX_CubeMap::Methods[] = {
+	KX_PYMETHODTABLE_NOARGS(KX_CubeMap, update),
+	{NULL, NULL} // Sentinel
+};
+
+PyAttributeDef KX_CubeMap::Attributes[] = {
+	KX_PYATTRIBUTE_RW_FUNCTION("viewpointObject", KX_CubeMap, pyattr_get_viewpoint_object, pyattr_set_viewpoint_object),
+	KX_PYATTRIBUTE_BOOL_RW("autoUpdate", KX_CubeMap, m_autoUpdate),
+	KX_PYATTRIBUTE_INT_RW("ignoreLayers", 0, (1 << 20) - 1, true, KX_CubeMap, m_ignoreLayers),
```

Comment:
Hi, I think there is an issue with this: http://pasteall.org/blend/index.php?id=43238